src/ess/reduce/time_of_flight/eto_to_tof.py

```diff
@@ -307,20 +305,26 @@ def __init__(self, lookup: sc.DataArray, distance_unit: str, time_unit: str):
         # monotonic, so we cannot use e.g. [-0.5, 0.5, 0.5, 1.5] in the case of
         # pulse_stride=2. Instead we use [-0.25, 0.25, 0.75, 1.25].
         base_grid = np.arange(float(lookup.sizes["pulse"]))
-        self._interpolator = RegularGridInterpolator(
-            (
-                np.sort(np.concatenate([base_grid - 0.25, base_grid + 0.25])),
-                lookup.coords["distance"].to(unit=distance_unit, copy=False).values,
-                lookup.coords["event_time_offset"]
-                .to(unit=self._time_unit, copy=False)
-                .values,
-            ),
-            np.repeat(
+        self._pulse_edges = np.sort(
+            np.concatenate([base_grid - 0.25, base_grid + 0.25])
+        )
```

Comment:
What is going on here? Where does the value of 0.25 come from? Can't you use the midpoints of the arange from above as edges? Or do you need to have gaps in between the bins?

Reply:
There is no need for gaps. I tried to explain it in the comment just above these lines. This has been in there for a while. The number 0.25 is just arbitrary. I could have chosen `[-0.49, 0.49, 0.51, 1.49]`. I just cannot have `[-0.5, 0.5, 0.5, 1.5]` (the double 0.5 in the middle). The first 2 values need to be on either side of 0, and the next 2 values on either side of 1, and so on.